[DA] 마이박스 응모 상세 API 연동 및 탭바 이슈 수정 및 결과 확인 화면에 실제 데이터 반영

AhnSangHee opened the pull request:
# 개요
- 🔗  이슈링크 : resolve https://github.com/mash-up-kr/GGiriGGiri_iOS/issues/142

# 변경사항

## 💻 작업 내용
- 결과 확인 > 결과 확인 페이지로 이동. 결과 확인 버튼 enable. 이후 결과 확인 버튼은 해당 결과로 바뀌어야함. (당첨 또는 꽝)
- 꽝 > 꽝 버튼은 disable. 카드 클릭 시 꽝 페이지 이동
- 당첨 > 당첨 버튼은 disable. 카드 클릭 시 당첨 페이지 이동
- 응모 중 > 응모 중 버튼은 disable. 카드 클릭 시 디테일 페이지 이동
- 탭 바 스크롤 문제 수정
- 응모 상세 API 연동
- 결과 확인 화면에 실제 데이터 반영

## 📱 스크린샷
- 현재 서버에서 당첨 기프티콘에 대한 카테고리 정보를 주지 않아서 현재는 임시로 카페/디저트 아이콘을 표시하도록 했습니다.
- 당첨 결과 화면에서 나오는 기프티콘 아이콘은 rotateImage가 아닌 똑바로 된 이미지로 변경해야합니다. 추후 변경하겠습니다!

마이박스 화면|결과 확인 화면
---|---
<img width="563" alt="스크린샷 2022-08-17 오전 2 08 46" src="https://user-images.githubusercontent.com/95578975/184938193-a465fb85-4557-4e21-974d-76be664c8254.png">|<img width="563" alt="스크린샷 2022-08-17 오전 2 11 22" src="https://user-images.githubusercontent.com/95578975/184938640-cbad3ba1-9837-4579-899e-76cb270a8f63.png">

AhnSangHee committed:
:bulb: 마이박스 화면 카드에 delegate 추가 (#142)

- 응모 BOX 카드만 다음 화면으로 이동하도록 구현
- 등록 BOX 카드는 이벤트 X

AhnSangHee committed:
:bulb: 마이박스 스와이프 이벤트 제거

AhnSangHee committed:
:bulb: 마이박스 탭 바 이슈 해결 (#142)

AhnSangHee committed:
:bulb: 마이박스 상세 응모 API 연동 및 결과 화면 실제 데이터 반영 (#142)

- 기프티콘 아이콘 클릭 시 실제 기프티콘 이미지 표시
- 기프티콘 데이터 받아와서 저장하도록 구현

AhnSangHee committed:
:bulb: 마이박스 결과 확인 버튼 이슈 수정 (#142)

- 마이박스 최초 결과 확인 화면에서 마이박스 화면으로 되돌아올경우, "결과확인" 버튼에서 "당첨/꽝"으로 버튼 변경

AhnSangHee committed:
:bulb: 등록 박스 셀 터치할 때 생기는 오류 수정 (#142)

AhnSangHee committed:
:memo: 주석 추가 및 불필요한 코드 제거 (#142)

AhnSangHee committed:
:recycle: 불필요한 코드 제거

kimkyunghun3 commented on Projects/App/Sources/MyBox/MyBoxListCollectionViewDelegate.swift:
```diff
@@ -0,0 +1,31 @@
+    weak var myBoxApplyListCardDelegate: MyBoxListCardDelegate?
+    
+    var type: MyBox
+    var id: Int = 0
```
요거 id는 변하는건가요?

kimkyunghun3 replied:
그리고 이거 외부에서 접근해야하나요?

kimkyunghun3 replied:
요거 안써서 지우고 접근제어가 필요한거 다 넣었습니다!

kimkyunghun3 commented on Projects/App/Sources/MyBox/View/MyBoxView.swift:
```diff
@@ -10,6 +10,7 @@ import UIKit
 
 import DesignSystem
 import SnapKit
+import RxRelay
```
조금 사소할수 있지만 통일성을 위해 S위에 R이 있으면 좋을거같네요!

kimkyunghun3 replied:
수정했습니다!

kimkyunghun3 reviewed (changes requested):
고생했습니다!

szzang-kr commented on Projects/App/Sources/MyBox/MyBoxListCollectionViewDelegate.swift:
```diff
@@ -24,6 +24,7 @@ final class MyBoxListCollectionViewDelegate: NSObject, UICollectionViewDelegate
     }
     
     func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if data.count == 0 { return }
```
어떤 이유에서 해당 코드로 오류가 수정이 되는걸까요?

ios-h replied:
현재 응모 BOX에서만 카드 클릭 시 다음 화면으로 이동하고, 등록 BOX는 터치 이벤트가 없습니다. 그래서 MyCollectionViewDataSource에서 type이 `applied`일 경우에만 data를 넣어주고 있고, 등록 BOX의 경우 data를 넣어주고 있지 않기 때문에 이 시점에서의 data 수가 0이기 때문에 저렇게 처리를 했었습니다! 근데 굳이 data수로 체크하기보다, type이 registered일 경우 return을 하는 게 더 좋을 것 같네요! [여기](https://github.com/mash-up-kr/GGiriGGiri_iOS/commit/f4e73d650aaa963c32d08387498cb1e32993c4b1)서 수정했습니다!

AhnSangHee replied:
현재 응모 BOX에서만 카드 클릭 시 다음 화면으로 이동하고, 등록 BOX는 터치 이벤트가 없습니다. 그래서 MyCollectionViewDataSource에서 type이 `applied`일 경우에만 data를 넣어주고 있고, 등록 BOX의 경우 data를 넣어주고 있지 않기 때문에 이 시점에서의 data 수가 0이여서 저렇게 처리를 했었습니다! 근데 굳이 data수로 체크하기보다, type이 `registered`일 경우 return을 하는 게 더 좋을 것 같네요! [여기](https://github.com/mash-up-kr/GGiriGGiri_iOS/commit/f4e73d650aaa963c32d08387498cb1e32993c4b1)서 수정했습니다!

szzang-kr commented on Projects/App/Sources/Result/ResultViewController.swift:
```diff
@@ -75,19 +101,23 @@ extension ResultViewController: ResultViewButtonDelegate {
-        }
-        UIImageWriteToSavedPhotosAlbum(image, self, #selector(gifticonImageSaved), nil)
-        completion(true)
+        guard let imageUrl = URL(string: "https://item.kakaocdn.net/do/e27878b2d7c30103f76bfb73833ffaab9f17e489affba0627eb1eb39695f93dd") else { return }
```
이미지 URL이 테스트 하면서 넣어둔 URL로 올려진것 같습니다.

확인 부탁드립니다 😓

kimkyunghun3 committed:
:recycle: 카테고리 맞게 icon 보이도록 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 불필요한 코드 제거 및 접근제어자 추가

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: reuse에 따른 이슈 해결

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/App/Sources/MyBox/View/MyBoxListCollectionViewCell.swift:
```diff
@@ -20,12 +20,11 @@ final class MyBoxListCollectionViewCell: UICollectionViewCell {
     static let reuseIdentifier = "MyBoxListCollectionViewCell"
     
-    private let disposeBag = DisposeBag()
+    private var disposable: Disposable?
```
cell 재사용시 binding 되어있는 disposable을 dispose 시켜주는것은 올바른 방향이나,

binding 해야하는것이 많아진다면 DisposeBag을 통해 Binding된 모든 내용을 dispose 시켜야 reuse시 문제가 발생하지 않습니다! 

[참고자료](https://makeeyaf.postype.com/post/9014926)

@kimkyunghun3 , @ios-h

kimkyunghun3 replied:
좋은 정보 감사합니다..! 일단은 하나라서 저렇게 할 수 있지만, 추후에 더 생긴다면 disposeBag으로 바꾸겠습니다!
RxSwift에서 자주 생기는 에러인거 같아서 잘 적어두겠습니다 :)

AhnSangHee committed:
:recycle: 피드백 반영

AhnSangHee committed:
Merge branch 'develop' into feature/sh/mybox
